dotnet/sdk-phase-1.md

```diff
@@ -0,0 +1,1022 @@
+GH pages, etc that don't align with this proposal necessarily. After some thought, it has been decided to:
+
+* Create a new https://github.com/temporalio/sdk-dotnet-incubating repo for the very first commits
+* After that repo has some code and API doc gen capability:
+  * Rename https://github.com/temporalio/sdk-dotnet to https://github.com/temporalio/legacy-dotnet (note the
+    intentionally absent "sdk" term for SEO reasons)
+    * Add README note on https://github.com/temporalio/legacy-dotnet pointing to
+      https://github.com/temporalio/sdk-dotnet
+    * Disable GH pages
+    * Archive it
+    * Consider deletion at some point in the future
+  * Rename https://github.com/temporalio/sdk-dotnet-incubating to https://github.com/temporalio/sdk-dotnet
+  * Update DNS for https://dotnet.temporal.io/ to point to vercel-based API docs page
+
+Doing the repo rename was deemed easier than going through manually closing issues, closing PRs, creating empty `main`
+branch, and keeping `master` branch around for those reading older code.
+
+## Namespace/Package
+
+* Will publish as `Temporalio`
+  * Need to reserve prefix `Temporalio` on NuGet, see
+    https://learn.microsoft.com/en-us/nuget/nuget-org/id-prefix-reservation
+* Namespace will be `Temporal`
+  * See discussion topics later about this being `Temporal.Sdk` instead
+
+## Workflow Definition
+
+Here's a definition for an other-language-defined workflow (from Python README):
+
+```csharp
+namespace MyNamespace;
+
+using System.Text.Json.Serialization;
+using Temporal.Workflow;
+
+public record GreetingInfo
+{
+    [JsonPropertyName("salutation")]
+    public string Salutation { get; init; } = "Hello";
+
+    [JsonPropertyName("name")]
+    public string Name { get; init; } = "<unknown>";
+}
+
+[Workflow]
+public interface IGreetingWorkflow
+{
+    static readonly IGreetingWorkflow Ref = Refs<IGreetingWorkflow>.Instance;
+
+    [WorkflowRun]
+    Task<string> RunAsync(GreetingInfo info);
+
+    [WorkflowSignal]
+    Task UpdateInfoAsync(GreetingInfo info);
+
+    [WorkflowSignal]
+    Task CompleteWithGreetingAsync();
+
+    [WorkflowQuery]
+    string CurrentGreeting();
+}
+```
+
+Notes:
+
+* We use regular .NET JSON serialization (see data conversion later)
+* Workflow attributes, and in the future workflow runtime features, are in the `Temporal.Workflow` namespace
+* We only support workflow interfaces right now because we don't have a workflow implementation yet
+  * For now, everything non-static must have some form of `[Workflow` attribute, and not have a default implementation
+* Every workflow interface must have the non-inheritable `Workflow` attribute
+  * Optional `Name` positional property can be given. Default is the unqualified type name with the "I" prefix removed
+    if it's an interface (which it has to be currently) and the next letter is also capital
+    * So what does `IDWorkflow` become? Yes, `DWorkflow` :-( Weighed as an acceptably rare cost for obvious benefit
```

Comment:
If I was forced to do this, I would just never infer a name ever. Workflow names not having little language-specific things like these prefixes be default I think is important. This would only make more typing for 99% of users who won't care for just one user that may be surprised we removed the `I` .NET expects to be the first char in some case where they want to reference it by string.

Reply:
> but IXXx would be rejected but that's probably taking it too far.

This would force people to violate the .NET standard. So `IIDWorkflow` or `IQRWorkflow` or whatever, valid names, would fail.

I think we can count on people always using the universally accepted `I` prefix for interfaces.